blockmanager.go

```diff
@@ -425,16 +429,20 @@ func (b *blockManager) handleDonePeerMsg(peers *list.List, p *peer) {
 
 	// Remove requested transactions from the global map so that they will
 	// be fetched from elsewhere next time we get an inv.
-	for k := range p.requestedTxns {
-		delete(b.requestedTxns, k)
+	for k, txPeer := range b.requestedTxns {
```

Comment:
An alternative idea here is to wrap the new `peer.Peer` in a btcd-specific peer that adds the peer-specific data it needs.  Something like:

```Go
type btcdPeer struct {
	*peer.Peer
	requestedTxns map[wire.ShaHash]struct{})
	...
}
```

However, I suspect the original suggestion is more efficient since it won't require the extra wrapping nor mapping of `peer.Peer` to btcdPeer that would have to be done in each of the callbacks.